hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -177,23 +171,23 @@
      * configured {@link javax.cache.integration.CacheLoader} might be called to retrieve
      * the value of the key from any kind of external resource.
      * <p>
-     * The resulting {@link com.hazelcast.core.ICompletableFuture} instance may throw a
+     * The resulting {@link com.hazelcast.core.CompletionStage} instance may throw a
      * {@link java.lang.ClassCastException} as the operations result if the {@link javax.cache.Cache}
      * is configured to perform runtime-type-checking, and the key or value types are incompatible
      * with those that have been configured for the {@link javax.cache.Cache}.
      *
      * @param key          The key whose associated value is to be returned.
      * @param expiryPolicy The custom expiry policy for this operation,
      *                     a null value is equivalent to {@link #getAsync(Object)}.
-     * @return ICompletableFuture retrieve the value assigned to the given key.
+     * @return CompletionStage retrieve the value assigned to the given key.
      * @throws java.lang.NullPointerException if given key is null.
      * @throws javax.cache.CacheException     if any exception
      *                                        happens while invoking the request, other exceptions are wrapped.
      * @throws IllegalStateException          if the cache is {@link #isClosed()}.
      * @see javax.cache.Cache#get(Object)
-     * @see com.hazelcast.core.ICompletableFuture
+     * @see com.hazelcast.core.CompletionStage
```

Comment:
Looks like it should be `java.util.concurrent.CompletionStage` or just `CompletionStage` and there are many such places.

Reply:
🤦‍♂

Reply:
Pushed a commit that fixes this. I also added javadoc that clarifies dependent actions execution conventions of non-async and async methods, since `CompletionStage` leaves this to be specified by the implementation. Probably this passage should be copied to the javadoc of each data structure that supports async operations.